drift.py

```diff
@@ -35,31 +36,29 @@ def analyze(self, commit):
                 # If concretization successful check the resulting concrete specs.
                 if out.returncode == SUCCESS:
                     concrete_spec = spack.spec.Spec().from_yaml(out.stdout)
-                    if abstract_spec in self.last and self.last[abstract_spec] != concrete_spec:
-                        # Construct Result
-                        diff = spack.cmd.diff.compare_specs(self.last[abstract_spec], concrete_spec, colorful=False)
-                        result = Result(
-                            abstract_spec[:verToken],
-                            abstract_spec[verToken+1:],
-                            commit.hash,
-                            # Waiting for spack diff to complete tags.
-                            diff['b_not_a'],
-                            str(commit.author_date))
-                        # Send result to drift-server
-                        send(result)
-                    # Save concrete spec as last spec
+                    # If the spec is the same move on.
+                    if abstract_spec not in self.last or self.last[abstract_spec] == concrete_spec:
+                        continue
+                    # If the spec is different check what is different and save as tags.
+                    diff = spack.cmd.diff.compare_specs(
```

Comment:
So let's see - this is setting A as the previous spec (in self.last) and B as the new concrete spec, so if you are grabbing "b_not_a" this is going to tell you what is in the new concrete spec but not the previous - I think that's what we want! I would do a sanity check on the first few runs to make sure the output makes sense.

Reply:
Just ran some tests and `b_not_a` does seem to be what we want!